### PR TITLE
Fix #669

### LINKED ISSUE #669: Null adjacency graph results in no temporal floodfill

### PR DESCRIPTION
**Pull request by nippoo**
This manually adds trivial connectivity graphs into the FloodFillDetector by passing a `channels` attribute into the `connected_components()` method,  fixing issue #669 where having a null adjacency graph, or leaving a channel out, results in no temporal floodfill being carried out.

### CODE REVIEW

**Comment by rossant** [comment]
travis is broken because of some setuptools bullsh*t -- see what I did at https://github.com/kwikteam/phy/commit/1b0d145965b173564cc2f67a64ed609c753312c7

**Comment by rossant** [comment]
also could you please add a test case for non regression?

File: phy/traces/detect.py [hunk 1]
```diff
@@ -190,6 +194,16 @@ def connected_components(weak_crossings=None,
...
+    if channels is None:
+        channels = []
+
+    # If the channels aren't referenced at all but exist in 'channels', add a
+    # trivial self-connection so temporal floodfill will work. If this channel
+    # is dead, it should be removed from 'channels'
+    for i in channels:
+        if not probe_adjacency_list.get(i):
+            probe_adjacency_list[i] = {i}
+
```
Comment (rossant) [inline phy/traces/detect.py]: `probe_adjacency_list = {i for i in channels if not probe_adjacency_list.get(i, None)}` ?

**Comment by nippoo** [comment]
thoughts on a good test for non-regression? in theory if the `mgraph` is non-null then we're all good so would a simple `assert mgraph` suffice?

the actual result of the floodfill failing is a bit more involved to test (you need to simulate data with two simultaneous strong threshold crossings and check that the connected components are merged correctly, but maybe this could also serve as a test for `join_size`)

can I just backport your travis fix?

**Comment by rossant** [comment]
I think it is worth making a very minimal toy failing test case fixed by this PR, so that we're sure that future versions of the code don't have the same code (notably upcoming phy 1.0). Look at the unit tests of flood fill, there are many toy examples.

**Comment by nippoo** [comment]
finally ready to merge @rossant

**Comment by rossant** [comment]
Thanks! I shall not forget to forward-port this fix to the other branch.

**Pull request merged**